server/src/main/java/org/spine3/server/aggregate/AggregateRepository.java

```diff
@@ -243,6 +231,55 @@ public void dispatch(Command request) throws IllegalStateException {
         commandStatusService.setOk(commandId);
     }
 
+    @SuppressWarnings("ChainOfInstanceofChecks")        // it's a rare case of handing an exception, so we are OK.
+    private A loadAndDispatch(I aggregateId, CommandId commandId, Message command, CommandContext context) {
+        final AggregateStorage<I> aggregateStorage = aggregateStorage();
+        A aggregate;
+
+        /**
+         * During the command dispatching and event applying, the original list of events may have been changed by
+         * other actors in the system.
+         *
+         * <p>To ensure the resulting {@code Aggregate} state is consistent with the numerous concurrent actor changes,
+         * the event count from the last snapshot should remain the same during the {@link AggregateRepository#load(Object)}
+         * and {@link Aggregate#dispatch(Message, CommandContext)}.
+         *
+         * <p>In case the new events are detected, {@code Aggregate} loading and {@code Command} dispatching is repeated
+         * from scratch.
+         */
+        Integer eventCountBeforeSave = null;
+        int eventCountBeforeDispatch;
+        do {
+            if (eventCountBeforeSave != null) {
+                log().warn("New events detected during the dispatch of the command {} " +
+                                   "Possible concurrent modification of {} {}" +
+                                   "Repeating the command dispatching.",
+                           command, getAggregateClass(), aggregateId);
+            }
+            eventCountBeforeDispatch = aggregateStorage.readEventCountAfterLastSnapshot(aggregateId);
+            aggregate = load(aggregateId);
+            try {
+                aggregate.dispatch(command, context);
+            } catch (RuntimeException e) {
+                final Throwable cause = e.getCause();
+                if (cause instanceof Exception) {
```

Comment:
At this point `ProcessManagerRepository` does not have any special error handling in scope of a command dispatching.

As a separate PR it makes sense to introduce such a handling. In addition, the `CommandStatusService` will probably be used for PM repositories.